core/amber/src/main/python/core/models/operator.py

```diff
@@ -96,6 +97,75 @@ def on_finish(self, port: int) -> Iterator[Optional[TupleLike]]:
         yield
 
 
+class BatchOperator(TupleOperatorV2):
+    """
+    Base class for batch-oriented operators. A concrete implementation must
+    be provided upon using.
+    """
+
+    DEFAULT_BATCH_SIZE: int = (
+        10  # if BATCH_SIZE is not initialized, then it will use DEFAULT_BATCH_SIZE
+    )
+    BATCH_SIZE: Optional[int] = DEFAULT_BATCH_SIZE  # must be a positive integer
```

Comment:
the type should be `int` instead of `Optional[int]`. We are disallowing `None` specificly